src/OpenSSL/SSL.py

```diff
@@ -699,8 +714,98 @@ def set_default_verify_paths(self):
 
         :return: None
         """
+        # This function will attempt to load certs from both a cafile and
+        # capath that are set at compile time. However, it will first check
+        # environment variables and, if present, load those paths instead
```

Comment:
`SSL_CTX_set_default_verify_paths` itself checks env vars. It's part of OpenSSL itself. The logic here should look like this:

1) Attempt to load via `SSL_CTX_set_default_verify_paths`. This will use the env vars specified by OpenSSL preferentially, but if they're not set (as they are not in most cases) then it will load the CA file and CA dir that were set at compile time.
2) If env vars are set we do not attempt to load any fallbacks, but if they are **not** we go to the fallback path.